File: 剑指offer/10-I斐波那契数列.go
func fib(n int) int {
    if n == 0 {
        return 0
    }
    if n == 1 {
        return 1
    }
    f1, f2 := 1, 0
    for i := 2; i <= n; i++ {
        f1, f2 = (f1 + f2) % (1e9+7), f1
    }
    return f1
}

File: 剑指offer/10-I斐波那契数列.py
class Solution:
    def fib(self, n: int) -> int:
        if n < 2:
            return n
        MOD = 10 ** 9 + 7
        f0, f1 = 0, 1
        for i in range(2, n + 1):
            f0, f1 = f1, f0 + f1
            # f1, f0 = f0 + f1, f1
            # print(i, f0, f1)

        return f1 % MOD
